### DISCUSSION TITLE
ESLint reports `Unable to resolve path to module` errors when using the modular admin SDK

### DISCUSSION THREAD

**Discussion by sceee** [firebase/firebase-admin-node#1359] [Q&A] [answered]
When using the new modular admin SDK in a Typescript project, eslint's `import/no-unresolved` rule throws a bunch of errors when linting:
```
Unable to resolve path to module 'firebase-admin/firestore'  import/no-unresolved
Unable to resolve path to module 'firebase-admin/auth'       import/no-unresolved
Unable to resolve path to module 'firebase-admin/storage'    import/no-unresolved
```
This happens for every occurrance.

The imports are just regular imports like this:
```typescript
import { DocumentReference, getFirestore, Timestamp } from 'firebase-admin/firestore'
import { getStorage } from 'firebase-admin/storage'
```

So this seems to be a bug as I'm using it pretty straightforward. Is anyone else running into this?

**Comment by hiranya911** [comment]
Does your code compile despite the lint errors?

**Comment by hiranya911** [comment, marked as answer]
It looks like `import/no-unresolved` is part of a custom ESLint plugin. I had to install the NPM package `eslint-plugin-import`, and add `plugin:import/recommended` to the ESLint config file for the rule to activate. With these config changes I'm able to reproduce the errors mentioned above. 

But I believe this is a bug/limitation in `eslint-plugin-import` (see https://github.com/benmosher/eslint-plugin-import/issues/1868). The code is indeed valid, and compiled by `tsc`. VSCode can also successfully resolve the module entry points.

**Reply by sceee** [reply]
Yeah I can confirm it compiles successfully by `tsc` so it seems to be the issue with eslint-plugin-import as you mentioned.

So I now just suppress the warning for `firebase-admin` using the following eslint rule:
```
'import/no-unresolved': [
      'error',
      {
        ignore: ['^firebase-admin/.+'],
      },
    ],
```

Thanks for your help!

**Reply by tettoffensive** [reply]
Thanks for this. I'm running into the same issue. Hopefully they support it someday.

**Reply by sergeushenecz** [reply]
Thanks it is working.
